app/src/main/java/org/kiwix/kiwixmobile/zimManager/ZimManageViewModel.kt

```diff
@@ -385,13 +385,20 @@ class ZimManageViewModel @Inject constructor(
     return None
   }
 
-  @Suppress("NoNameShadowing")
+  private fun updateNetworkStates() {
```

Comment:
Please fix the compilation error, see the screenshot below. 

Now `updateNetworkStates` method is coroutine-based and has the `Job` instead of `Disposable`. Additionally, include this job in `coroutineJobs` list in the `observeCoroutineFlows` method so that it can be cancel when viewModel is cleared to avoid memory leak, and unwanted resource usages.


![image](https://github.com/user-attachments/assets/51d46b47-03a7-4689-8379-81a77d5e36aa)